### PR TITLE
Issue #1 - Migration from pages router to app router

### PR DESCRIPTION
**Pull request by OneRandom1509**
# Migration to App Router

This pull request addresses issue #1 by migrating the Nex music player application from the "pages router" to the "app router." This change helps us take advantage of the latest Next.js features and improves the overall maintainability and performance of Nex. 

In this process, `_app.tsx` and `_document.tsx` have been merged into `layout.tsx` file in the root directory of the app folder, and `404.tsx` has been replaced with the new convention - `not-found.tsx`. Some components have been added with `use client;` directive in order to use the new `router()` router instance and `usePathame()` function from `next/navigation` instead of the old router instance from `next/router`, both of which can only be used in the client side.

In addition to the migration, the `SEO` page has been replaced with the new built-in `Metadata` API to export metadata easily inside the app router.

Every `page` inside the app router has been split into two files - `layout.tsx` and `page.tsx`. `layout.tsx` is where every client side operation takes place and `page.tsx` is where server side operations take place. For now, the only server side operation in `page.tsx` files is the export of metadata using Metadata API. A void function has been written in order to bypass the error of not exporting a react component in the file. This void function can be used for future server side features as well.

# Screenshots
The improved not-found page
localhost_3000_example

### CODE REVIEW

File: src/app/account/layout.tsx [hunk 1]
```diff
@@ -1,11 +1,9 @@
+'use client';
```
Comment (ThEditor) [inline src/app/account/layout.tsx]: Do not make pages client-side which can be server-side.
Try to implement pages in server-side if possible. If not, then convert them to client-side.

File: src/app/login/layout.tsx [hunk 2]
```diff
@@ -1,9 +1,11 @@
-import { useRouter } from 'next/router';
+'use client';
```
Comment (ThEditor) [inline src/app/login/layout.tsx]: Though the file uses "useEffect", all it does is redirect the user if they're already signed in. Try to stay on server-side and redirect.
Refer: [NextJS Redirect Stackoverflow](https://stackoverflow.com/questions/58173809/next-js-redirect-from-to-another-page)

**Review by ThEditor — CHANGES_REQUESTED** [review]
The code seems to default towards client-side components which is against the app router convention, hence, try to default to server-side pages and server-side components. If it's not possible to implement it as server-side then try to go for `'use client'`

File: src/app/account/layout.tsx (hunk 1 above)
Comment (OneRandom1509) [reply src/app/account/layout.tsx]: Alright, I'll make the updates in the upcoming PR

File: src/app/login/layout.tsx (hunk 2 above)
Comment (OneRandom1509) [reply src/app/login/layout.tsx]: Updates coming up in the next PR

**Comment by OneRandom1509** [comment]
The new commits have made all the pages to be server-side and components to be server-side, except the sidebar and the carousel as it needs the user interactivity in order for it to work. All the above commented required changes have been made in a proper manner too.

**Review by ThEditor — APPROVED** [review]
After discussing this further on discord, I'm merging these changes.

**Pull request merged**